src/main/java/seedu/staffsnap/logic/parser/ApplicantBookParser.java

```diff
@@ -64,7 +72,15 @@ public Command parseCommand(String userInput) throws ParseException {
             return new DeleteCommandParser().parse(arguments);
 
         case ClearCommand.COMMAND_WORD:
-            return new ClearCommand();
+            if (IsConfirmed) {
+                return new ClearCommand();
+            } else {
+                logger.finer("This user input caused a ParseException: " + userInput);
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            }
+        case ConfirmationCommand.COMMAND_WORD:
+            IsConfirmedNext = true;
+            return new ConfirmationCommand();
```

Comment:
Maybe look into abstracting away the control flow logic to the command execution.
For example, we can shift the check of `isConfirmed` into `ClearCommand().execute` so that we let it throw the error.

But perhaps this will have to do for v1.2.